x-pack/plugins/search_playground/server/utils/get_chat_params.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OPENAI_CONNECTOR_ID } from '@kbn/stack-connectors-plugin/common/openai/constants';
+import { ActionsClientChatOpenAI, ActionsClientLlm } from '@kbn/elastic-assistant-common/impl/llm';
+import { v4 as uuidv4 } from 'uuid';
+import { BEDROCK_CONNECTOR_ID } from '@kbn/stack-connectors-plugin/common/bedrock/constants';
+import { PluginStartContract as ActionsPluginStartContract } from '@kbn/actions-plugin/server';
+import { KibanaRequest, Logger } from '@kbn/core/server';
+import { BaseLanguageModel } from '@langchain/core/language_models/base';
+import { Prompt } from '../../common/prompt';
+
+export const getChatParams = async (
+  {
+    connectorId,
+    model,
+    prompt,
+    citations,
+  }: { connectorId: string; model?: string; prompt: string; citations: boolean },
+  {
+    actions,
+    request,
+    logger,
+  }: {
+    actions: ActionsPluginStartContract;
+    logger: Logger;
+    request: KibanaRequest;
+  }
+): Promise<{ chatModel: BaseLanguageModel; chatPrompt: string }> => {
+  const abortController = new AbortController();
+  const abortSignal = abortController.signal;
+  const actionsClient = await actions.getActionsClientWithRequest(request);
+  const connector = await actionsClient.get({ id: connectorId });
+  let chatModel;
+  let chatPrompt;
+
+  if (connector.actionTypeId === OPENAI_CONNECTOR_ID) {
```

Comment:
would a `switch` be better here as we already have more that 2 connector types.

Should we be handling other connector `actionTypeId` values here already?

Reply:
> Should we be handling other connector actionTypeId values here already?

only some of the connectors that we support today: bedrock & openai. We ignore the rest of the kibana connectors.  Theres actually only two types of genai connectors also.

updated the code to use Switch for readability. For now I break on default. In future as more genai connectors get added, we can highlight the ones which were ignored.